--- a/drivers/spi/spi-falcon.c
+++ b/drivers/spi/spi-falcon.c
@@ -10,6 +10,7 @@
 #include <linux/device.h>
 #include <linux/platform_device.h>
 #include <linux/spi/spi.h>
+#include <linux/spi/spi-mem.h>
 #include <linux/delay.h>
 #include <linux/of.h>
 #include <linux/of_platform.h>
@@ -387,6 +388,15 @@ static int falcon_sflash_xfer_one(struct
 	return 0;
 }
 
+static const char *falcon_sflash_get_name(struct spi_mem *mem)
+{
+	return "sflash";
+}
+
+static const struct spi_controller_mem_ops falcon_sflash_mem_ops = {
+	.get_name = falcon_sflash_get_name,
+};
+
 static int falcon_sflash_probe(struct platform_device *pdev)
 {
 	struct falcon_sflash *priv;
@@ -404,6 +414,7 @@ static int falcon_sflash_probe(struct pl
 	host->setup = falcon_sflash_setup;
 	host->transfer_one_message = falcon_sflash_xfer_one;
 	host->dev.of_node = pdev->dev.of_node;
+	host->mem_ops = &falcon_sflash_mem_ops;
 
 	ret = devm_spi_register_controller(&pdev->dev, host);
 	if (ret)
